src/pages/UserTable/DropDownUserTable.tsx

```diff
@@ -44,130 +48,134 @@ const DropDown = (props: Props) => {
     pageX,
     pageY,
     showDropdown,
+    inActiveTab,
     onDelete,
     onChange,
-    user,
+    selectedUser,
+    selectedUserRoles,
     currentUser,
     canView
   } = props;
+
   const [showEditModal, setShowEditModal] = useState(false);
   const [visibleModalDegree, setVisibleModalDegree] = useState<boolean>(false);
   const [showCityModal, setShowCityModal] = useState<boolean>(false);
   const [showRegionModal, setShowRegionModal] = useState<boolean>(false);
   const [showClubModal, setShowClubModal] = useState<boolean>(false);
 
-  const [superAdmin, setsuperAdmin] = useState<boolean>(false);
+  const [superAdmin, setSuperAdmin] = useState<boolean>(false);
   const [governingBodyHead, setGoverningBodyHead] = useState<boolean>(true);
+
+  const [currentUserAdminRoles, setCurrentUserAdminRoles] = useState<Array<AdminRole>>([]);
+  const [selectedUserAdminRoles, setSelectedUserAdminRoles] = useState<Array<AdminRole>>([]);
+
+  const [canDelete, setCanDelete] = useState<boolean>(false);
   const [canChangeCityAdministration, setCanChangeCityAdministration] = useState<boolean>(false);
   const [canChangeClubAdministration, setCanChangeClubAdministration] = useState<boolean>(false);
   const [canChangeRegionAdministration, setCanChangeRegionAdministration] = useState<boolean>(false);
   const [canChangeGoverningBodyAdministration, setCanChangeGoverningBodyAdministration] = useState<boolean>(false);
   const [canChangeUserAccess, setCanChangeUserAccess] = useState<boolean>(false);
   const [canAddDegree, setCanAddDegree] = useState<boolean>(false);
 
+  const [chainOfAccessibility, setChainOfAccessibility] = useState<IDropdownItem>();
+  
+  //Takes only those roles, which can access User Table and 
+  //writes them in array in descending order (as in AdminRole enum)
+  const setUserAdminRoles = (allUserRoles: Array<string>) => {
+
+    const userAdminRoles: Array<AdminRole> = new Array<AdminRole>();
+
+    if (allUserRoles?.includes(Roles.Admin)) {
+      userAdminRoles.push(AdminRole.Admin);
+    }
+    if (allUserRoles?.includes(Roles.GoverningBodyHead)) {
+      userAdminRoles.push(AdminRole.GoverningBodyHead);
+    }
+    if (allUserRoles?.includes(Roles.OkrugaHead)) {
+      userAdminRoles.push(AdminRole.RegionHead);
+    }
+    if (allUserRoles?.includes(Roles.OkrugaHeadDeputy)) {
+      userAdminRoles.push(AdminRole.RegionHeadDeputy);
+    }  
+    if (allUserRoles?.includes(Roles.CityHead)) {
+      userAdminRoles.push(AdminRole.CityHead);
+    }
+    if (allUserRoles?.includes(Roles.CityHeadDeputy)) {
+      userAdminRoles.push(AdminRole.CityHeadDeputy);
+    }
+    if (allUserRoles?.includes(Roles.KurinHead)) {
+      userAdminRoles.push(AdminRole.ClubHead);
+    }      
+    if (allUserRoles?.includes(Roles.KurinHeadDeputy)) {
+      userAdminRoles.push(AdminRole.ClubHeadDeputy);
+    }
+
```

Comment:
You should perform string intersection to detect common roles instead of those ifs

Reply:
Ok, thx

Reply:
Done